test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpHandlerResourceV3ProviderTests.cs

```diff
@@ -124,5 +127,122 @@ static IEnumerable<DelegatingHandler> GetDelegatingHandlers(HttpMessageHandler h
                 }
             }
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task TryCreate_WhenCertificateValidationIsConfigured_HandlerShouldReflectConfiguration(bool disableCertificateValidation)
+        {
+            // Arrange
+            Mock<IProxyCache> proxyCache = new();
+            proxyCache.Setup(pc => pc.GetProxy(It.IsAny<Uri>())).Returns((IWebProxy)null);
+
+            PackageSource packageSource = new(_testPackageSourceURL, "source")
+            {
+                DisableTLSCertificateValidation = disableCertificateValidation
+            };
+
+            SourceRepository sourceRepository = new(packageSource, Array.Empty<INuGetResourceProvider>());
+
+            HttpHandlerResourceV3Provider target = new(proxyCache.Object);
+
+            // Act
+            var result = await target.TryCreate(sourceRepository, CancellationToken.None);
+
+            // Assert
+            result.Item1.Should().BeTrue();
+            HttpHandlerResourceV3 resource = (HttpHandlerResourceV3)result.Item2;
+            resource.Should().NotBeNull();
+
+            HttpClientHandler clientHandler = resource.ClientHandler;
+
+            if (disableCertificateValidation)
+            {
+                clientHandler.ServerCertificateCustomValidationCallback.Should().NotBeNull();
+                var callbackResult = clientHandler.ServerCertificateCustomValidationCallback.Invoke(null, null, null, SslPolicyErrors.RemoteCertificateChainErrors
+                    & SslPolicyErrors.RemoteCertificateNameMismatch
+                    & SslPolicyErrors.RemoteCertificateNotAvailable
+                    & SslPolicyErrors.None);
+                callbackResult.Should().BeTrue();
+            }
+            else
+            {
+                clientHandler.ServerCertificateCustomValidationCallback.Should().BeNull();
+            }
```

Comment:
Conditionals in a test typically indicate that we're testing different inputs or state and they should be different tests or a data driven test with different inputs. Please consider breaking this test into multiple tests.